### PR TITLE
Add assert to as<int32_t>() for dirty higher bits.

### PR DESCRIPTION
**Pull request by axic**
Missed in #612.

### CODE REVIEW

File: test/unittests/execute_numeric_test.cpp [hunk 1]
```diff
@@ -264,7 +264,8 @@ TEST(execute_numeric, i32_mul)
 
 TEST(execute_numeric, i32_div_s)
```
Comment (axic) [inline test/unittests/execute_numeric_test.cpp:265]: This is especially annoying as we never had a proper 32-bit tests.

**Comment by axic** [comment]
@chfast @gumb0 oh I forgot to ping y'all for reviews. I'm not sure what's left, but probably we could extend the 32-bit tests if you have some ideas for more edge cases to cover.

**Review by gumb0 — APPROVED** [review]

**Review by chfast — APPROVED** [review]

**Review by chfast — APPROVED** [review]
